specs/specs.js

```diff
@@ -256,41 +289,65 @@ describe('Fingerprint2', function () {
         done()
       })
     })
-    describe('customFunction option', function () {
-      it('concatinates the current keys with the customFunction output', function (done) {
-        function customFunction () {
-          return 'RANDOM_STRING'
+    describe('extraComponents option', function () {
+      it('uses extraComponents', function (done) {
+        function customFunction (done) {
+          done('RANDOM_STRING')
         }
+
         var spy = jasmine.createSpy('customFunction', customFunction).and.callThrough()
         var fp = new Fingerprint2({
-          'customFunction': spy
+          extraComponents: [{key: 'customFunction', getData: spy}]
         })
         fp.get(function (result, keys) {
           expect(spy).toHaveBeenCalled()
           done()
         })
       })
-      it('with customKey option changes the key', function (done) {
+      it('its key is used', function (done) {
         var fp = new Fingerprint2({
-          customFunction: function () {
-            return 'RANDOM_STRING'
-          },
-          customKey: 'TEST_STRING'
+          extraComponents: [
+            {
+              key: 'TEST_STRING',
+              getData: function customFunction (done) {
+                done('RANDOM_STRING')
+              }
+            }
+          ]
         })
         fp.get(function (_, components) {
           expect(getComponent(components, 'TEST_STRING')).toEqual('RANDOM_STRING')
           done()
         })
       })
+
+      it('safely introduce a new component even if it thorws', function (done) {
+        var fp = new Fingerprint2({
+          extraComponents: [
+            {
+              key: 'my key',
+              getData: function customFunction (done) {
+                throw 'unstable component'
+              }
+            }
+          ]
+        })
+        fp.get(function (_, components) {
+          // if we arrive here, it means the rest of the fingerprint was not disturbed
+          expect(components).toBeDefined()
+          done()
+        })
+      })
     })
 
     if (!onPhantomJs) {
       describe('enumerate devices fingerprint', function () {
         it('checks enumerate devices fingerprint', function (done) {
-          var fp2 = new Fingerprint2(); 
+          var fp2 = new Fingerprint2({excludes: []})
```

Comment:
Is passing `{excludes: []}` equivalent to passing `{}` or even not passing anything at all?

Reply:
passing {excludes: []} is not the same. It overwrites the default options, which has a couple of excludes

Reply:
passing {} or nothing will use the default options

Reply:
I see!